src/controller/AutoCommissioner.cpp

```diff
@@ -174,6 +176,35 @@ CHIP_ERROR AutoCommissioner::SetCommissioningParameters(const CommissioningParam
     }
     mParams.SetCSRNonce(ByteSpan(mCSRNonce, sizeof(mCSRNonce)));
 
+    if (params.GetDSTOffsets().HasValue())
+    {
+        ChipLogProgress(Controller, "Setting DST offsets from parameters");
+        size_t size = params.GetDSTOffsets().Value().size();
```

Comment:
nit: consider:

```cpp
const size_t size = std::min(params.GetDSTOffsets().Value().size(), kMaxSupportedDstStructs);
```

For shorter code

Reply:
I even wonder about:

```cpp
const size_t size = std::min(params.GetDSTOffsets().Value().size(), kMaxSupportedDstStructs);
auto list = app::DataModel::List<app::Clusters::TimeSynchronization::Structs::DSTOffsetStruct::Type>(mDstOffsetsBuf, size);

CopySpanToMutableSpan(params.GetDSTOffsets().Value().SubSpan(0, size), list);

mParams.SetDSTOffsets(list);
```

not sure if more readable though :/

Reply:
Yes, min certainly is the correct function here, and I somehow managed to forget it existed. Which is just ... wow.

Weirdly, CopySpanToMutableSpan is only available for byte and char spans, so I can't use that here. I'm not sure WHY we don't have a templated version, but we don't.